timestep.c

```diff
@@ -65,11 +64,15 @@ init_timebins(double TimeInit)
     PM.length = 0;
     PM.Ti_kick = All.Ti_Current;
     PM.start = All.Ti_Current;
-    update_active_timebins(0);
 }
 
-int is_timebin_active(int i) {
-    return TimeBinActive[i];
+int is_timebin_active(int i, inttime_t current) {
+    /*Bin 0 is always active and at time 0 all bins are active*/
+    if(i == 0 || current == 0)
+        return 1;
+    if(current % dti_from_timebin(i) == 0)
```

Comment:
perhaps it shal be == dti_from_timebin(i) / 2?

Reply:
No, because dti_from_timebin is the delta t?

Reply:
I see. 
Our scheme is D f k t k. 

- before t, due to k, the particle has synced ti_kick and ti_drift with ti_current.
Therefore it is safe to move particle to a new time step size. 

- before f, due to  D ti_drift and ti_current are the same.
- f needs to know if a particle is active. At this time ti_kick must be earlier than ti_drift. the particle is active iif ti_drift % dti == 0.

A particle is active == force will be computed, kick will be applied

Reply:
Right!